src/daemon/rpmostreed-os.c

```diff
@@ -77,6 +85,131 @@ sysroot_changed (RpmostreedSysroot *sysroot,
+    {
+      g_ptr_array_add (actions, "org.projectatomic.rpmostree1.repo-refresh");
+    }
+  else if (g_strcmp0 (method_name, "Deploy") == 0)
+    {
+      g_ptr_array_add (actions, "org.projectatomic.rpmostree1.deploy");
+    }
+  else if (g_strcmp0 (method_name, "Upgrade") == 0)
+    {
+      g_ptr_array_add (actions, "org.projectatomic.rpmostree1.upgrade");
+    }
+  else if (g_strcmp0 (method_name, "Rebase") == 0)
+    {
+      g_ptr_array_add (actions, "org.projectatomic.rpmostree1.rebase");
+    }
+  else if (g_strcmp0 (method_name, "SetInitramfsState") == 0)
+    {
+      g_ptr_array_add (actions, "org.projectatomic.rpmostree1.set-initramfs-state");
+    }
+  else if (g_strcmp0 (method_name, "Cleanup") == 0)
+    {
+      g_ptr_array_add (actions, "org.projectatomic.rpmostree1.cleanup");
+    }
+  else if (g_strcmp0 (method_name, "Rollback") == 0 ||
+           g_strcmp0 (method_name, "ClearRollbackTarget") == 0)
+    {
+      g_ptr_array_add (actions, "org.projectatomic.rpmostree1.rollback");
+    }
+  else if (g_strcmp0 (method_name, "PkgChange") == 0)
+    {
+      g_ptr_array_add (actions, "org.projectatomic.rpmostree1.install-uninstall-packages");
+    }
+  else if (g_strcmp0 (method_name, "UpdateDeployment") == 0)
+    {
+      g_autoptr(GVariant) modifiers = g_variant_get_child_value (parameters, 0);
+      g_autoptr(GVariant) options = g_variant_get_child_value (parameters, 1);
+      g_auto(GVariantDict) modifiers_dict;
+      g_auto(GVariantDict) options_dict;
+      g_variant_dict_init (&modifiers_dict, modifiers);
+      g_variant_dict_init (&options_dict, options);
+      const char *refspec =
+        vardict_lookup_ptr (&modifiers_dict, "set-refspec", "&s");
+      const char *revision =
+        vardict_lookup_ptr (&modifiers_dict, "set-revision", "&s");
+      g_autofree char **install_pkgs =
+        vardict_lookup_ptr (&modifiers_dict, "install-packages", "^a&s");
+      g_autofree char **uninstall_pkgs =
+        vardict_lookup_ptr (&modifiers_dict, "uninstall-packages", "^a&s");
+      g_autoptr(GVariant) install_local_pkgs =
+        g_variant_dict_lookup_value (&modifiers_dict, "install-local-packages",
+                                     G_VARIANT_TYPE("ah"));
+      gboolean no_pull_base =
+        vardict_lookup_bool (&options_dict, "no-pull-base", FALSE);
+
+      if (refspec != NULL)
+        g_ptr_array_add (actions, "org.projectatomic.rpmostree1.rebase");
+      else if (revision != NULL)
+        g_ptr_array_add (actions, "org.projectatomic.rpmostree1.deploy");
+      else if (!no_pull_base)
+        g_ptr_array_add (actions, "org.projectatomic.rpmostree1.upgrade");
+
+      if (install_pkgs != NULL || uninstall_pkgs != NULL)
+        g_ptr_array_add (actions, "org.projectatomic.rpmostree1.install-uninstall-packages");
+
+      if (install_local_pkgs != NULL && g_variant_n_children (install_local_pkgs) > 0)
+        g_ptr_array_add (actions, "org.projectatomic.rpmostree1.install-local-packages");
```

Comment:
Hmm, we'll need another check here for the override related modifiers, under e.g. a new `org.projectatomic.rpmostree1.override` action? Not sure if it's worth splitting into multiple actions just yet. They're experimental, but don't use the experimental interface like livefs.